sdk/dotnet/Pulumi/Deployment/Deployment.Runner.cs

```diff
@@ -43,7 +43,10 @@ public void RegisterTask(string description, Task task)
             {
                 lock (_taskToDescription)
                 {
-                    _taskToDescription.Add(task, description);
+                    if (!_taskToDescription.ContainsKey(task))
+                    {
+                        _taskToDescription.Add(task, description);
+                    }
```

Comment:
why was this changed?  this indicates that we're adding hte task multiple times.  that's likely not a good thing.

1. if this is correct, we need to doc when/why this happens.
2. if this is correct, it's simpler to just do `_taskToDescription[task] = description`.

Note: the `.Add` was very intentional on my part to catch bugs.  i.e. i wanted it to crash if we somehow double registered a task.

Reply:
Gotcha. I changed it because we return `Task.CompletedTask` from mocks and they obviously collide here.
`Task.CompletedTask` is not guaranteed to be the same instance, so we can't really compare to it. Would it be safe to ignore completed tasks here?

Reply:
Fixed by merging the master